application/btsieve/src/ethereum/queries/mod.rs

```diff
@@ -1,7 +1,82 @@
-pub mod block_query;
-pub mod event_query;
-pub mod transaction_query;
+pub mod block;
+pub mod event;
+pub mod transaction;
 
-pub use self::{
-    block_query::BlockQuery, event_query::EventQuery, transaction_query::TransactionQuery,
+pub use self::{block::BlockQuery, event::EventQuery, transaction::TransactionQuery};
+use crate::route_factory::Error;
+use ethereum_support::{
+    web3::{transports::Http, Web3},
+    Transaction, TransactionId, TransactionReceipt,
 };
+use ethereum_types::{clean_0x, H256};
+use futures::Future;
+
+fn to_h256<S: AsRef<str>>(tx_id: S) -> Option<H256> {
+    let tx_id = tx_id.as_ref();
+
+    match hex::decode(clean_0x(tx_id)) {
+        Ok(bytes) => Some(H256::from_slice(&bytes)),
+        Err(e) => {
+            warn!("skipping {} because it is not valid hex: {:?}", tx_id, e);
+            None
+        }
+    }
+}
+
+#[derive(Serialize, Debug)]
+#[serde(untagged)]
+pub enum PayloadKind {
+    Id {
+        id: H256,
+    },
+    Transaction {
+        transaction: Box<Transaction>,
+    },
+    Receipt {
+        receipt: Box<TransactionReceipt>,
+    },
+    TransactionAndReceipt {
+        transaction: Box<Transaction>,
+        receipt: Box<TransactionReceipt>,
+    },
+}
+
+// TODO: Investigate under which circumstances the ethereum node fails to return
```

Comment:
❓ would such investigation be worth it? What would you do with the result.
I feel like we cannot really answer this and hence questioning the need for a TODO

Reply:
Resolved!